third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSTaskIdPropagate.cpp

```diff
@@ -24,6 +24,16 @@ namespace ttng = ::mlir::triton::nvidia_gpu;
 namespace mlir {
 
 int doTaskIdPropagate(triton::FuncOp &funcOp) {
+  // Convert ttg.partition to async_task_id
+  funcOp.walk([&](mlir::Operation *op) {
+    if (auto attr = op->getAttrOfType<IntegerAttr>("ttg.partition")) {
+      int64_t idx = attr.getInt();
+      assert(idx >= 0);
+      setAsyncTaskIds(op, idx);
+      op->removeAttr("ttg.partition");
```

Comment:
Can we delete the code handling ttg.partition that we've added? (e.g. in `get/removeAsyncTaskId` and `TMEMAlloc1D`)